### PR TITLE
Added 'getCountries' and 'getTerritories' methods to 'CLocale' class, enh #3479

### PR DESCRIPTION
**Pull request by ua2004**

### CODE REVIEW

**Comment by samdark** [comment]
#3479

File: framework/i18n/CLocale.php [hunk 1]
```diff
@@ -469,4 +469,31 @@ public function getTerritory($id)
...
+	 * @return array list of local display names of all countries.
+	 * @since 1.1.17
+	 */
+	public function getCountries()
+	{
+		$countries = array();
+		foreach($this->_data['territories'] as $code => $country)
+		{
+			//zz is an unknown region
+			if(($code != 'zz') && preg_match('/[a-z]{2}/', $code))
```
Comment (samdark) [inline framework/i18n/CLocale.php]: What's `preg_match` check for?

Comment (ua2004) [inline framework/i18n/CLocale.php]: `$this->_data['territories'] ` contains all territories - countries as well as world regions and continents. Region codes contain only digits
```
'001' => 'World',
'002' => 'Africa',
```
meanwhile country codes consist of only 2 letters.
```
'ar' => 'Argentina',
'au' => 'Australia',
```
All we have to do is to filter numerical codes leaving only letter codes, i.e. countries.

Comment (samdark) [inline framework/i18n/CLocale.php]: I see.

**Comment by samdark** [comment]
Looks OK except minor formatting issues.

**Pull request closed without merging**